lib/sidekiq_unique_jobs/lock/while_executing.rb

```diff
@@ -17,21 +17,23 @@ def lock(_scope)
       # @return [Boolean] report success
       # @raise [SidekiqUniqueJobs::LockTimeout] when lock fails within configured timeout
       def execute(callback)
-        jid = @lock.lock
-        if jid == @item[JID_KEY]
-          callback&.call
-          unlock(:server)
+        locked = @lock.lock(@calculator.lock_timeout)
```

Comment:
It is possible the corresponding tests for this file is broken and needs fixing. I haven't had time to go through it all. Wanted to organise the tests and the files better.